### DISCUSSION TITLE
probleme de connexion socket esp32

### DISCUSSION THREAD

**Discussion by william7-prog** [micropython/micropython#10730] [ESP32]
I have this error message (comment solve this please):
>>> %Run -c $EDITOR_CONTENT
Borne wifi :  Tchanyap
network config: ('192.168.100.42', '255.255.255.0', '192.168.100.1', '192.168.100.1')
Traceback (most recent call last):
  File "<stdin>", line 23, in <module>
AttributeError: 'NoneType' object has no attribute 'bind'
>>> 
  
with this python code:

'''import network
import socket

wlan = network.WLAN(network.STA_IF)
wlan.active(True)
wlan.isconnected()

ssid= ""
password= ""

print("Borne wifi : ", ssid)
if not wlan.isconnected():
    print('connecting to network...')
    wlan.connect(ssid,password)
    while not wlan.isconnected():
        pass
if wlan.isconnected():
    print('network config:', wlan.ifconfig())

serveursocket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
serveursocket.bind(('', 80))
serveursocket.listen(5)

def web_page():
    html = """<!DOCTYPE HTML>
        <html>
        <head>
        <meta name="viewport" content="width=device-width, initial-scale=1">
        </head>
        <body>
            <h2> ESP32 WebServer</h2>
            <p>Bienvenue sur l'ESP32</p>
        </body>
        </html>"""
    return html
'''

**Comment by robert-hh** [comment]
Even if the formatting of your code example was lost, it looks ok. I would write the wait loop as:
```
    import time
    while not wlan.isconnected():
        time.sleep_ms(10)
```
The `import time` should be further up in the code. 

Notes about posting: Please enclose code in lines containing three backticks ```. Then it will be shown properly formatted. And please use the English language.

**Reply by william7-prog** [reply]
d'accord merci pour les conseils. dorenavant j'ecrirai mon code en anglais.

**Reply by robert-hh** [reply]
The code should be written in Python, and the name of the variables can be any. But the natural language text besides the code ought to be English. You can use a translator like `deepl translator` or `Google translate` which work pretty well.
